session/time_tracker.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package session
+
+import "time"
+
+// TimeTracker tracks elapsed time from the beginning of the session
+// it's passive (no internal go routines) and simply encapsulates time operation: now - startOfSession expressed as duration
+type TimeTracker struct {
+	started   bool
+	startTime time.Time
+	getTime   func() time.Time
+}
+
+// NewTracker initializes TimeTracker with specified monotonically increasing clock function (usually time.Now is enough - but we do DI for test sake)
+func NewTracker(getTime func() time.Time) TimeTracker {
+	return TimeTracker{
+		getTime: getTime,
+	}
+}
+
+// StartTracking starts tracking the time
+func (tt *TimeTracker) StartTracking() {
+	tt.started = true
+	tt.startTime = tt.getTime()
+}
+
+// Elapsed gets the total duration of time that has passed since we've started
+func (tt TimeTracker) Elapsed() time.Duration {
+	if !tt.started {
```

Comment:
We avoid using `started` field. 
We can check [time.IsZero()](https://golang.org/pkg/time/#Time.IsZero) here: 
```suggestion
	if tt.startTime.IsZero() {
```

Reply:
Actually there are slight differences what time IsZero actually reports and what we rely on. More about that - golang monotonic vs wall clock on google. And here I prefer to have explicit flag for two main reasons:
1. It shows intent or reason more clearly - return zero duration if not started
2. IsZero function doesn't say anything about monotonic clock values - and we heavily rely on that

Reply:
Update: It's probably save to use IsZero here, as it also checks nsec part of monotonic clock - however there are no guarantees it will change in the future. Anyway I would stick for first argument. @vkuznecovas its your call